src/test/java/com/google/sps/TimingPropertiesTest.java

```diff
@@ -190,4 +190,108 @@ public void getNewQuestion_validParamters() {
-}
\ No newline at end of file
+ 
+    @Test 
+    //Test if giveUserPoints is given null Entity value as a parameter
+    public void giveUserQuizTakenPoints_nullEntity() {
+        QuizTimingPropertiesUtils timing_properties_test = new QuizTimingPropertiesUtils();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity user = null;
+ 
+        boolean actual = timing_properties_test.giveUserQuizTakenPoints(true, user, datastore);
+        Assert.assertEquals(false, actual);
+    }
+ 
+    @Test 
+    //Test if giveUserPoints is given null datastore value as a parameter
+    public void giveUserQuizTakenPoints_nullParameters() {
+        QuizTimingPropertiesUtils timing_properties_test = new QuizTimingPropertiesUtils();
+        DatastoreService datastore = null;
+        Entity user = new Entity("user");
+ 
+        boolean actual = timing_properties_test.giveUserQuizTakenPoints(true, user, datastore);
+        Assert.assertEquals(false, actual);
+    }
+ 
+    @Test
+    //Tests of userQuizStatus is false
+    public void giveUserQuizTakenPoints_falseUserQuizStatus(){
+        QuizTimingPropertiesUtils timing_properties_test = new QuizTimingPropertiesUtils();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity user = new Entity("user");
+ 
+        boolean actual = timing_properties_test.giveUserQuizTakenPoints(false, user, datastore);
+        Assert.assertEquals(false, actual);
+    }
+ 
+    @Test
+    //Tests if user does not have a score property 
+    public void giveUserQuizTakenPoints_notScoreProperty() {
+        QuizTimingPropertiesUtils timing_properties_test = new QuizTimingPropertiesUtils();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        
+        Entity user = new Entity("user");
+        user.setProperty("quiz_timestamp", 159430944365L);
+        user.setProperty("userID", 12345);
+        datastore.put(user);
+ 
+        boolean actual = timing_properties_test.giveUserQuizTakenPoints(true, user, datastore);
+        Assert.assertEquals(false, actual);
+    }
+ 
+    @Test
+    //Test if user was given points if their initial value was not zero
+    public void giveUserQuizTakenPoints_actuallyAddedPoints_notStartAtZero(){
+        QuizTimingPropertiesUtils timing_properties_test = new QuizTimingPropertiesUtils();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        
+        Entity user = new Entity("user");
+        user.setProperty("quiz_timestamp", 159430944365L);
+        user.setProperty("userID", 12345);
+        int score_value = 15;
+ 
+        user.setProperty("score", score_value);
+        datastore.put(user);
+ 
+        boolean actual = timing_properties_test.giveUserQuizTakenPoints(true, user, datastore);
+        if(((int) user.getProperty("score")) == (score_value + 20) ) {
+            Assert.assertEquals(actual, true);
+        }
```

Comment:
So in this case, we actually don't want to test the result of giveUserQuizTakenPoints(). We want to assert on assertEquals(entity.getProperty("score"), 35);